packages/sit-onyx/src/utils/playwright.tsx

```diff
@@ -0,0 +1,177 @@
+  Array.from({ length: rows + 1 })
+    .map((_, ri) =>
+      Array.from({ length: columns + 1 })
+        .map((_, ci) => {
+          if (ri === 0 && ci === 0) return ".";
+          if (ri === 0) return `colHeadline${ci - 1}`;
+          if (ci === 0) return `rowHeadline${ri - 1}`;
+          else return `case${(ri - 1) * columns + ci - 1}`;
+        })
+        .join(" "),
+    )
+    .map((v) => `"${v}"`) // every row is surrounded by "
+    .join("\n"); // every column has its own line
+
+type ComponentStates = Readonly<Record<string, ReadonlyArray<string>>>;
+type CaseBuilder<S extends ComponentStates> = (c: Permutation<S>, i: number) => JSX.Element;
+
+/**
+ * Generates a test case in the grid and places it accordingly.
+ */
+const createTestCases = <S extends ComponentStates>(
+  testCases: Permutation<S>[],
+  caseBuilder: CaseBuilder<S>,
+) =>
+  testCases.map((tC, i) => (
+    <div
+      style={{ gridArea: `case${i}` }}
+      key={i}
+      // Handle common cases
+      class={{
+        [`pw-${tC["focusState"]}`]: tC["focusState"],
+        "onyx-use-optional": tC["state"] === "optional",
+      }}
+    >
+      {caseBuilder(tC, i)}
+    </div>
+  ));
+
+/**
+ * Generates the jsx for the screenshot matrix test.
+ * It uses a grid with named areas to place all test cases, the row and column headlines.
+ */
+const generateScreenshotMatrix = <S extends ComponentStates>(
+  states: S,
+  caseBuilder: CaseBuilder<S>,
+) => {
+  const [rowStates] = Object.values(states);
+  const colStates = generatePermutations(Object.fromEntries(Object.entries(states).slice(1))).map(
+    (e) => Object.values(e),
+  );
+
+  const columns = colStates.length;
+  const rows = rowStates.length;
+  const template = buildGridTemplate(rows, columns);
+
+  const testCases = generatePermutations(states);
+
+  return (
+    <main
+      data-testid="screenshot-root"
+      style={{
+        fontFamily: "var(--onyx-font-family-mono)",
+        padding: "1rem",
+        display: "grid",
+        gap: "0.75rem",
+        "grid-template-areas": template,
+      }}
+    >
+      {
+        // Create column headlines
+        colStates.map((state, i) => (
+          <div
+            key={i}
+            style={{
+              gridArea: `colHeadline${i}`,
+              display: "flex",
+              flexDirection: "column",
+              writingMode: "vertical-lr",
+              textOrientation: "mixed",
+            }}
+          >
+            {state.map((s) => (
+              <span key={s}>{s}</span>
+            ))}
+          </div>
+        ))
+      }
+      {
+        // Create row headlines
+        rowStates.map((state, i) => (
+          <div key={i} style={{ gridArea: `rowHeadline${i}` }}>
+            {state}
+          </div>
+        ))
+      }
+      {createTestCases(testCases, caseBuilder)}
+    </main>
+  );
+};
+
+type TestArg = Parameters<Parameters<typeof test>[1]>[0];
+
+export const createMatrixScreenshot =
```

Comment:
Some JSDoc would be nice so its easier when other devs use this in the future

Reply:
done